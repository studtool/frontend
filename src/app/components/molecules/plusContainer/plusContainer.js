import React, {Component} from 'react';
import {AddButton} from '../../atoms/addButton/addButton.js';

export class PlusContainer extends Component {
    constructor(props) {
        super(props);

        this.handleAdd = this.handleAdd.bind(this);
    }

    handleAdd(e){
        console.log("Click");
    }

    render() {
        const {
            qa = false,
            data = {},
            handleAdd = this.handleAdd,
            children = null,
        } = this.props;

        return (
            <div data-qa={qa ? 'plus-container' : void 0}>
                <AddButton handleClick={handleAdd}/>
                {children}
            </div>
        );
    };
}
