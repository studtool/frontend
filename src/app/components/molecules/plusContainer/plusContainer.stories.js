import * as React from 'react';
import {storiesOf} from '@storybook/react';
import {PlusContainer} from 'App/components/molecules/plusContainer/plusContainer';
import {FileSystemContainer} from 'App/components/molecules/fileSystemContainer/fileSystemContainer';

storiesOf('plus container ', module)
    .add('default', () => (
        <PlusContainer>
            <FileSystemContainer>
                Название файла
            </FileSystemContainer>
        </PlusContainer>
    ))
    .add('with qa', () => (
        <PlusContainer qa={true}/>
    ))
