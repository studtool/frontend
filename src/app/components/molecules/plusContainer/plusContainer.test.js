import React from 'react';
import {PlusContainer} from './plusContainer';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

test('file System Container test', () => {
    const stubSubmit = () => {};
    const stubChange = () => {};
    const data = {
        email__errorMessage: '',
        password__errorMessage: '',
        passwordRepeat__errorMessage: '',
    };

    const tree = renderer.create(
        <PlusContainer
            handleSubmit={stubSubmit}
            handleChange={stubChange}
            data={data}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
