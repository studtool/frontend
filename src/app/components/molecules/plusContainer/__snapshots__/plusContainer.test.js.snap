// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`file System Container test 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
